pkg/tsdb/prometheus/resource/resource.go

```diff
@@ -0,0 +1,125 @@
+package resource
```

Comment:
Is this trying to conform to some interface defined somewhere else? It feels a bit non ideal that we have a nice API wrapper and we are not using it. At the same time when we move to the streaming parsing we will lose that. @toddtreece or is there any plan to copy similar interface from the go prometheus client?